extensions/markdown-language-features/src/markdownEngine.ts

```diff
@@ -25,6 +25,7 @@ const pluginSourceMap: MarkdownIt.PluginSimple = (md): void => {
 			if (token.map && token.type !== 'inline') {
 				token.attrSet('data-line', String(token.map[0]));
 				token.attrJoin('class', 'code-line');
+				token.attrJoin('dir', 'auto');
```

Comment:
Is this required or is `dir` inherited by child elements?

Reply:
@mjbvz 

It is required. Because `dir="auto"` just looks at the first paragraph and apply direction to all the child element.
Childs just inherit `rtl` or `ltr`. Childs doesn't inherit `auto` and it is not so smart.

I think we can only add `dir="auto"` to the `code-line` classes (childs). But for `markdown-body` class is optional.